odoo/fields.py

```diff
@@ -4531,7 +4531,8 @@ def flush():
                             to_create.append(dict(command[2], **{inverse: record.id}))
                         allow_full_delete = False
                     elif command[0] == Command.UPDATE:
-                        comodel.browse(command[1]).write(command[2])
+                        prefetch_ids = recs[self.name]._prefetch_ids
+                        comodel.browse(command[1]).with_prefetch(prefetch_ids).write(command[2])
```

Comment:
If your update commands target different subset of fields, it will fetch too much (more) in modified? But I don't think this is a problem anyway (maybe we should have a multi_write (values_list) to cover this case the most efficiently - completely out of scope).